### DISCUSSION TITLE
Feature request: Allow defining full body in HTTPException

### DISCUSSION THREAD

**Discussion by plaa** [fastapi/fastapi#10741] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
raise HTTPException(status_code=status.HTTP_409_CONFLICT, detail=body_response)
```

### Description

The `HTTPException` is a useful way of raising errors to the HTTP level, however, the `detail` argument is always wrapped inside the `detail` field in the response.  While wrapping in the `detail` field is logical for simple textual responses, there's no way to provide the full top-level response body.  

I propose adding an optional `content` (or similar) argument for `HTTPException` which would provide the full content for `JSONResponse` in the exception handler.  This is a trivial, backward-compatible change that would make `HTTPException` more versatile.

Currently the only options I know of for error responses is explicitly returning a `JSONResponse` (which doesn't work from nested calls) or implementing a separate custom HTTP exception + exception handler, which is essentially the same as what I propose above.

I can create a PR for this if it would likely be accepted.

### Operating System

macOS

### FastAPI Version

0.104.1

### Pydantic Version

2.4.2

### Python Version

3.10.13

**Comment by Kludex** [comment, marked as answer]
This will not be accepted, because there are other ways to achieve the same.

You can create your custom exception, and create an exception handler, for example, or you can directly return the response you want.

Also, this request should have been in Starlette.

**Reply by plaa** [reply]
May I ask why is it better to require writing a custom exception + exception handler instead of providing the functionality out-of-the-box?  I was rather surprised there was no option for providing the body and the custom exception seems like a workaround for missing functionality.
